### PR TITLE
Skip mounting for iSCSI volumes and write iSCSI target instead

### PR DESCRIPTION
**Pull request by LLParse**

### CODE REVIEW

File: csi/node_server.go [hunk 1]
```diff
@@ -3,24 +3,80 @@ package csi
...
+func (ns *NodeServer) getISCSITargetByVolumeName(name string) (string, error) {
+	selector, err := getVolumeSelector(name)
+	if err != nil {
+		return "", err
+	}
+	opts := metav1.ListOptions{
+		LabelSelector: selector.String(),
 	}
+	// FIXME where are we going to get namespace from..?
+	engineList, err := ns.lhClient.LonghornV1alpha1().Engines("longhorn-system").List(opts)
```
Comment (LLParse) [inline csi/node_server.go]: unacceptable; need to set POD_NAMESPACE in longhorn-csi-plugin pods

File: csi/node_server.go [hunk 2]
```diff
@@ -3,26 +3,89 @@ package csi
...
+	lhClient, err := lhclientset.NewForConfig(config)
+	if err != nil {
+		logrus.Warningf("NewNodeServer failed to create Longhorn client: %v", err)
+	}
+
+	namespace := os.Getenv(types.EnvPodNamespace)
+	if namespace == "" {
+		logrus.Warnf("Cannot detect pod namespace, environment variable %v is missing, "+
+			"using default namespace", types.EnvPodNamespace)
+		namespace = corev1.NamespaceDefault
```
Comment (yasker) [inline csi/node_server.go]: Thinking about it, we should just error out. Shouldn't fall back to default.

Can you fix https://github.com/rancher/longhorn-manager/blob/master/controller/controller_manager.go#L32 as well? As another commit.

File: csi/node_server.go [hunk 3]
```diff
@@ -34,6 +97,18 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 
 	targetPath := req.GetTargetPath()
 
+	if frontend, ok := req.VolumeAttributes["frontend"]; ok && frontend == "iscsi" {
```
Comment (yasker) [inline csi/node_server.go]: If the volume exists before and being used without `frontend` field in the Kubernetes API request, this code path won't be triggered. We should check the volume property using Longhorn API directly here.

**Comment by yasker** [comment]
Close this PR for now.

**Pull request closed without merging**